app/models/service_template_transformation_plan.rb

```diff
@@ -0,0 +1,62 @@
+class ServiceTemplateTransformationPlan < ServiceTemplate
+  def request_class
+    ServiceTemplateTransformationPlanRequest
+  end
+
+  def request_type
+    "transformation_plan"
+  end
+
+  def add_vms(vm_ids)
+    vm_ids.each { |vm_id| add_vm(vm_id) }
+    save!
```

Comment:
Can we load the VM objects and use the existing `add_resource` logic of ServiceTemplate instead of creating our own `add_vm` method?  You may not require the `save!` if you make this change.

Reply:
This is purposely to by-pass the constraint that a VM can belong to only one service/service_template.

Reply:
Does it make more sense to override the `enforce_single_service_parent` method for this service type?
https://github.com/ManageIQ/manageiq/blob/master/app/models/mixins/service_mixin.rb#L113

For now it could just skip it and in the future be used to control if a VM can be in multiple migration plans.